source/Scrapers/TeamCityCompatibleAgentsScraper.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Prometheus.Client;
+using Serilog;
+using TeamCitySharp;
+
+namespace TeamCityBuildStatsScraper.Scrapers
+{
+    class TeamCityCompatibleAgentsScraper : BackgroundService
+    {
+        readonly IMetricFactory metricFactory;
+        readonly IConfiguration configuration;
+        readonly HashSet<(string buildTypeId, string buildId)> seenBuildsNoAgents = new();
+
+        public TeamCityCompatibleAgentsScraper(IMetricFactory metricFactory, IConfiguration configuration, ILogger logger)
+            : base(logger.ForContext("Scraper", nameof(TeamCityCompatibleAgentsScraper)))
+        {
+            this.metricFactory = metricFactory;
+            this.configuration = configuration;
+        }
+
+        protected override TimeSpan DelayBetweenScrapes => TimeSpan.FromSeconds(60);
+
+        protected override async Task Scrape(CancellationToken stoppingToken)
+        {
+            await Task.CompletedTask;
+
+            var teamCityToken = configuration.GetValue<string>("TEAMCITY_TOKEN");
+            var teamCityUrl = configuration.GetValue<string>("BUILD_SERVER_URL");
+            var useSSL = configuration.GetValue<bool>("USE_SSL");
+            var teamCityClient = new TeamCityClient(teamCityUrl, useSSL);
+
+            teamCityClient.ConnectWithAccessToken(teamCityToken);
+
+            var queuedBuilds = teamCityClient.BuildQueue
+                .GetFields("count,build(id,waitReason,buildTypeId,queuedDate,statistics(property,value,name),compatibleAgents(count,agent(id)))")
```

Comment:
```suggestion
                .GetFields("count,build(id,waitReason,buildTypeId,queuedDate,compatibleAgents(count,agent(id)))")
```
As discussed on Zoom; minimum fields here helps with performance on the TeamCity side.

Reply:
I removed this, thanks